pkgs/games/etlegacy/default.nix

```diff
@@ -0,0 +1,124 @@
+{ stdenv
+, lib
+, makeWrapper
+, writeScriptBin
+, fetchgit
+, fetchFromGitHub
+, fetchzip
+, fetchurl
+, runCommand
+, cmake
+, git
+, glew
+, SDL2
+, zlib
+, minizip
+, libjpeg
+, curl
+, lua5_4
+, libogg
+, libtheora
+, freetype
+, libpng
+, sqlite
+, openal
+, unzip
+, cjson
+}:
+
+let
+  version = "2.81.1";
+  pkgname = "etlegacy";
+  lua = lua5_4;
+  mirror = "https://mirror.etlegacy.com";
+  fetchAsset = {asset, sha256}: fetchurl
+    { url = mirror + "/etmain/" + asset;
+    inherit sha256;
+    };
+  pak0 = fetchAsset
+    { asset = "pak0.pk3";
+    sha256 = "712966b20e06523fe81419516500e499c86b2b4fec823856ddbd333fcb3d26e5";
+    };
+  pak1 = fetchAsset
+    { asset = "pak1.pk3";
+    sha256 = "5610fd749024405b4425a7ce6397e58187b941d22092ef11d4844b427df53e5d";
+    };
+  pak2 = fetchAsset
+    { asset = "pak2.pk3";
+    sha256 = "a48ab749a1a12ab4d9137286b1f23d642c29da59845b2bafc8f64e052cf06f3e";
+    };
+  fakeGit = writeScriptBin "git" ''
+    #! ${stdenv.shell} -e
+    if [ "$1" = "describe" ]; then
+      if [ -r VERSION.txt ]; then
```

Comment:
it was not possible to replace the git call directly with the nix version (often done with substituteInPlace) ? some packages react differently when git is not available too.

Reply:
Okay, yes that's much easier - I don't know why I didn't think of that. :)

```
fakeGit = writeScriptBin "git" ''
    #! ${stdenv.shell} -e
    if [ "$1" = "describe" ]; then
      echo "${version}"
    fi
  '';
```

Reply:
Ah, after re-reading your comment I realised what you meant. I guess that would be adding something like this:

```nix
patchPhase = ''
  substituteInPlace cmake/ETLVersion.cmake --replace old new
'';
```

The relevant part of the CMAKE file is copied below. I could replace the `${GIT_DESCRIBE}` and `${GIT_DESCRIBE_TAG}` with the `"${version}"`, but I think git failing might break the build.

```
execute_process(COMMAND git describe --abbrev=7
                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
                OUTPUT_STRIP_TRAILING_WHITESPACE
                OUTPUT_VARIABLE GIT_DESCRIBE)
execute_process(COMMAND git describe --abbrev=0
                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
                OUTPUT_STRIP_TRAILING_WHITESPACE
                OUTPUT_VARIABLE GIT_DESCRIBE_TAG)

if(GIT_DESCRIBE)
        set(ETL_CMAKE_VERSION ${GIT_DESCRIBE})
        set(ETL_CMAKE_VERSION_SHORT ${GIT_DESCRIBE_TAG})
```

We do already have `git` as a `nativeBuildInput` - I guess we can't just let it actually run?

Reply:
@teto for the time being I've used the simplified `fakeGit` script above, as I didn't see a robust way to use `substituteInPlace` on the cmake file.